src/compiler.js

```diff
@@ -106,7 +106,7 @@ export function create(options = {}) {
     if (rule.classNames)
       fns.push(
         failText(({ element: { props } }) => {
-          let className = props && props.className
+          let className = props && props.className && props.className.toString()
```

Comment:
It might be better to do something a bit more implicit like `'' + className` just in case there isn't a toString (maybe `valueOf` is defined instead), at the very least it won't throw.